### PR TITLE
do the same in jax as in jax-jit for getting results in execute_fwd

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
The jax code was wrongly iterating over `[[res1, res2, ...], jacs]` instead of `[res1, res2, ...]` but it slipped by undetected because you'd need 3 or more tapes to be executed to hit an IndexError.

**Description of the Change:**
Remove the unnecessary tracing stuff and match what jax-jit does.

**Benefits:**
Code now works as expected

**Possible Drawbacks:**
It's a bit hacky, but it gets the job done.

### CODE REVIEW

**Review by albi3ro — COMMENTED** [review]
Also, still need to fix the
```
TypeError: Custom JVP rule must produce primal and tangent outputs with equal shapes and dtypes, but got:
  primal float32[] for tangent float32[1,0]
```
that occurs whenever we take a derivative.

File: tests/interfaces/test_jax.py [hunk 1]
```diff
@@ -150,8 +156,8 @@ def cost(a):
         a = jax.numpy.array([0.1, 0.2])
         cost(a)
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply tests/interfaces/test_jax.py:157]: ahhh. I [commented out that test](https://github.com/PennyLaneAI/pennylane/blob/5a9fd249b717dae80ec7f48dd2efe57deee38188/tests/interfaces/default_qubit_2_integration/test_jax_default_qubit_2.py#LL280C11-L280C11) when copying over to jax with a note. I might have just said that to not have to figure out the problem? but we're not testing it for jax : (

[the remark this answers is not in the document]
Comment (albi3ro) [reply tests/interfaces/test_jax.py:157]:
```

def cost(params):
    tape1 = qml.tape.QuantumScript(
        [qml.RY(params[0], 0), qml.RX(params[1], 0)],
        [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))],
        shots=shots,
    )

    tape2 = qml.tape.QuantumScript(
        [qml.RY(np.array(0.5), 0)],
        [qml.expval(qml.PauliZ(0))],
        shots=shots,
    )

    tape3 = qml.tape.QuantumScript(
        [qml.RY(params[0], 0), qml.RX(params[1], 0)],
        [qml.expval(qml.PauliZ(0))],
        shots=shots,
    )
    return qml.execute([tape1, tape2], device, **execute_kwargs)

params = jnp.array([0.1, 0.2])
x, y = params

res = cost(params)


assert np.allclose(res[0][0], np.cos(x) * np.cos(y))
assert np.allclose(res[0][1], 1)
assert np.allclose(res[1], np.cos(0.5))
assert np.allclose(res[2], np.cos(x) * np.cos(y))
```

File: tests/interfaces/test_jax.py [hunk 2]
```diff
@@ -127,31 +127,46 @@ def cost(a, device):
...
                     "method": "adjoint_jacobian",
                     "use_device_state": True,
                 },
-            )[0]
+            )
 
         a = jax.numpy.array([0.1, 0.2])
-        cost(a)
+        res = cost(a)
+        print(res)
```
Comment (albi3ro) [inline tests/interfaces/test_jax.py:160]:
```suggestion
```

**Review by albi3ro — APPROVED** [review]
Thanks for this quick fix.

We can keep trying to track down the a fix for the differentiation, but that can be a later fix.

**Pull request merged**